### ISSUE TITLE
undefined is not a constructor (evaluating 'this.documentResponsiveListener()') 

### ISSUE THREAD

**Issue by wheelerlaw** [jasmine/jasmine#1262]
I'm using a `ConfirmDialog` component from the PrimeNG library in my angular 2 app. The component works fine when testing it out in a browser, but for some reason, using the component makes Karma/Jasmine explode. 

     Error: Error in ./ReturnFileParameterComponent class ReturnFileParameterComponent - inline template:2:4 caused by: undefined is not a constructor (evaluating 'this.documentResponsiveListener()') in C:/Users/wlaw/Desktop/ens_fits_ui/karma-shim.js (line 41475)
            ngOnDestroy@C:/Users/wlaw/Desktop/ens_fits_ui/karma-shim.js:71991:45 <- webpack:///~/primeng/components/confirmdialog/confirmdialog.js:140:0
            ngOnDestroy
            destroyInternal
            destroy@C:/Users/wlaw/Desktop/ens_fits_ui/karma-shim.js:41076:34 <- webpack:///~/@angular/core/bundles/core.umd.js:12361:0
            ...

In the `component.spec.ts` file, I imported both the dialog service and module needed for the dialog component to work:

    import {
        ...
        ConfirmDialogModule,
        ConfirmationService
    } from 'primeng/primeng';

    describe('Return File Parameter Component Test', () => {
    
        beforeEach(() => {
            TestBed.configureTestingModule({
                declarations: [ReturnFileParameterComponent, HtmlUserFriendlyTimePipe, Html24HourDatePipe, DayOfWeekPipe],
                providers: [
                    ...
                    ConfirmationService,
                    ...
                ],
                imports: [
                    ...
                    ConfirmDialogModule
                ]
            });
            TestBed.compileComponents();
        });

    ...

        it('should call add on a new Return File Parameter and submit it to the service', inject(
            [ReturnFileParameterComponent, AuthHelper],
            (component: ReturnFileParameterComponent, authHelper: AuthHelper) => {
                let fixture: ComponentFixture<ReturnFileParameterComponent> =
                    getTestBed().createComponent(ReturnFileParameterComponent);
    
                fixture.componentInstance.ngOnInit()
                    .then(fixture.detectChanges)
                    .then(function () {
                        let newParam = fixture.componentInstance.add();
                        expect(fixture.componentInstance.selectedParam).toBe(newParam);
                        expect(fixture.componentInstance.editParam).toBe(true);
                    });
            })
        );
    });

But almost every test fails. 

Here is the beginning portion of the template file:

    <div class="ui-grid-row" id="return-parameter-table">
        <h5>Return File Parameters</h5>
        <p-growl [value]="msgs" life="3000"></p-growl>
        <p-confirmDialog header="Confirmation" icon="fa fa-question-circle" width="425"></p-confirmDialog>
    ...

I'm using `jasmine-core` 2.2.5 with `karma` 1.4.0 and `karma-jasmine` 1.1.0. I've seen this issue before without using Karma that was a result of some dependency injection issues, but here I can't quite figure it out since it looks like all the correct dependencies are imported. 

I'm running the tests with PhantomJS, but I've also tried Firefox and Chrome to see if it was a browser issue, but the error persists. 

So basically I have no idea why its failing the tests but I get absolutely no problems using it in a browser. Any help would be greatly appreciated.

**Comment by slackersoft** [comment]
You are more likely to get quicker responses from the community for "How to use jasmine?" questions and a history of other solutions on the [jasmine-js group](https://groups.google.com/group/jasmine-js). We try to keep jasmine's github issues list focused on bugs and feature requests for jasmine itself.

Closing. Thanks for using Jasmine!
